content/02-model/01-train-and-deploy-model.qmd

```diff
@@ -0,0 +1,300 @@
+    date >= train_start_date,
+    date <= train_end_date
+  ) %>%
+  distinct() %>%
+  collect()
+
+print(glue::glue(
+  "The model will be trained on data from {start} to {end} ",
+  "({num_obs} observations). ",
+  start = min(train_data$date),
+  end = max(train_data$date),
+  num_obs = scales::comma(nrow(train_data)),
+))
+
+# Test data split.
+test_data <- all_days %>%
+  filter(date > train_end_date) %>%
+  distinct() %>%
+  collect()
+
+print(glue::glue(
+  "The model will be tested on data from {start} to {end} ",
+  "({num_obs} observations). ",
+  start = min(test_data$date),
+  end = max(test_data$date),
+  num_obs = scales::comma(nrow(test_data)),
+))
+```
+
+## Train the model
+
+### Data preprocessing
+
+Define a recipe to clean the data.
+
+```{r define_recipe}
+# Define a recipe to clean the data.
+recipe_spec <- 
+  recipe(n_bikes ~ ., data = train_data) %>% 
+  step_dummy(dow) %>%
+  step_integer(id, date)
+
+# Preview the cleaned training data.
+recipe_spec %>% 
+  prep(train_data) %>% 
+  bake(head(train_data)) %>%
+  glimpse()
+```
+
+### Fit model
+
+Fit a random forest model:
+
+```{r fit_model}
+model_spec <- 
+  rand_forest() %>%
+  set_mode("regression") %>%
+  set_engine("ranger")
+
+model_workflow <- 
+  workflow() %>%
+  add_recipe(recipe_spec) %>%
+  add_model(model_spec)
+
+model_fit <- fit(model_workflow, data = train_data)
+model_fit
+```
+
+## Model evaluation
+
+```{r evaluate_model}
+predictions <- predict(model_fit, test_data)
+
+results <- test_data %>%
+  mutate(preds = predictions$.pred)
+
+oos_metrics(results$n_bikes, results$preds)
+```
+
+## Model deployment
+
+### `vetiver`
+
+Create a `vetiver` model object.
+
+```{r create_vetiver_model}
+model_name <- "bike_predict_model_r"
+pin_name <- glue("sam.edwardes/{model_name}")
+
+v <- vetiver_model(
+  model_fit, 
+  model_name,
+  versioned = TRUE,
+  save_ptype = train_data %>%
+    head(1) %>%
+    select(-n_bikes),
+  metadata = list(
+    train_dates = c(min(train_data$date), max(train_data$date)),
+    test_dates = c(min(test_data$date), max(test_data$date))
+  )
+)
+
+v
+```
+
+### `pins`
+
+Save the model as a *pin* to RStudio Connect:
+
+```{r pin_model}
+# Use RStudio Connect as a board.
+board <- pins::board_rsconnect(
+  server = Sys.getenv("CONNECT_SERVER"),
+  key = Sys.getenv("CONNECT_API_KEY"),
+  versioned = TRUE
+)
+
+# Write the model to the board.
+board %>%
+ vetiver_pin_write(vetiver_model = v)
+```
+
+Update the pin permissions and metadata on Connect:
+
+```{r pin_metadata}
+# Set content permissions
+client <- connectapi::connect(
+  server = Sys.getenv("CONNECT_SERVER"),
+  api_key = Sys.getenv("CONNECT_API_KEY")
+)
+
+# Get the content
+content <- client %>%
+  connectapi::content_item("0273207f-0f1f-44cb-845e-408f9dae2dd6")
+
+content$update(title = "Bike Predict (R) - Model - Pinned Model")
+
+# Make the app visible to the world!
+content$update(access_type = "all")
+
+# Define the contents collaborators
+collaborators <- c(
```

Comment:
We should remove @xuf12 and me from auto collaborators since we are publishing this repo publicly.

Reply:
I wanted to include you as collaborators so that you can make changes to the content if needed. E.g. I am on vacation and something goes wrong with the content. If I am the only collaborator only I can make changes.

What do you think?

Reply:
I removed all of the collaborator logic in 618731f.